File: src/g1ipmlinkedlist/Python/llist_node.py
class LinkedListNode:
    def __init__(self, data, next_node=None):
        self.data = data
        self.next = next_node


def new_linked_list_node(data, next_node=None):
    return LinkedListNode(data, next_node)


def init_linked_list_node(data):
    return LinkedListNode(data)

File: src/g1ipmlinkedlist/Go/llist_node.go
package main

// LinkedListNode defines the structure of a node in the linked list.
type LinkedListNode struct {
	data int
	next *LinkedListNode
}

// NewLinkedListNode initializes a new node with given data and next pointer.
func NewLinkedListNode(data int, next *LinkedListNode) *LinkedListNode {
	node := new(LinkedListNode)

	node.data = data
	node.next = next

	return node
}

// InitLinkedListNode initializes a new node with given data and no next pointer.
func InitLinkedListNode(data int) *LinkedListNode {
	node := new(LinkedListNode)

	node.data = data
	node.next = nil

	return node
}
